worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocumentEngine.java

```diff
@@ -63,10 +64,26 @@ public void markupDocument(final Document document, final List<HashConfiguration
         final MarkupWorkerConfiguration config = document.getApplication().getService(ConfigurationSource.class)
             .getConfiguration(MarkupWorkerConfiguration.class);
         final DataStore dataStore = document.getApplication().getService(DataStore.class);
+        final Codec codec = document.getApplication().getService(Codec.class);
+
+        String addEmailHeadersDuringMarkupOverrideStr = document.getCustomData("addEmailHeadersDuringMarkup");
```

Comment:
The JIRA's ambiguous on that;
"This should be configurable, i.e. allow for content that contains headers for the current email and one that doesn't."
There may be certain files that should have headers extracted and some that don't. Making it possible to override via customData allows for this rather than locking a worker to a specific implementation.
It also has benefits for testing as in the yaml we can pass the option as enabled or disabled via custom data. When it is only set via an environment vairable or config the tests are locked into that single mode.